### PR TITLE
Feature/knotx#167 update readme

### PR DESCRIPTION
**Pull request by Skejven**
README update
---

I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,118 +3,67 @@
...
+### Stability and responsiveness
+**Knot.x** uses asynchronous programming principles which allows it to process a large number of requests using a single thread.
+Asynchronous programming is a style promoting the ability to write non-blocking code (no thread pools).
+The platform stays responsive under heavy and varying load and is designed to follow [Reactive Manifesto](http://www.reactivemanifesto.org/) principles.
+
+### Loose coupling
+Relies on asynchronous message-passing to establish a boundary between system components that ensures 
+loose coupling, isolation and location transparency. Base **Knot.x** component is called [Knot](https://github.com/Cognifide/knotx/wiki/Knot).
 
 We care a lot about speed and that is why we built it on [Vert.x](http://vertx.io/), known as one of the leading frameworks for performant, event-driven applications.
```
Comment (tomaszmichalak) [inline README.md]: I believe that this sentence is more relevant to responsiveness.

File: README.md [hunk 2]
```diff
@@ -3,118 +3,67 @@
...
 
+### Scalability
+Various scaling options are available to suit client needs and help in cost optimization. Using a 
+simple concurrency model and message bus **Knot.x** can be scaled within a single host or cluster of 
+servers
+
+
+## What problems does Knot.x solve?
+**Knot.x** joins content from multiple sources on Repository and Data Ingestion levels to render dynamic pages:
```
Comment (tomaszmichalak) [inline README.md]: maybe?
"Knot.x assembles static and dynamic content to produce page with dynamic data in very performant manner."

File: README.md [hunk 3]
```diff
@@ -3,118 +3,67 @@
...
+
+<p align="center">
+  <img src="https://github.com/Cognifide/knotx/blob/master/documentation/src/main/wiki/assets/knotx-overview.png?raw=true"
+         alt="Knot.x overview"/>
+</p>
+
+- Thanks to **Knot.x** your system may have several templates (pages) source, all joined in **Knot.x** using [Repository](https://github.com/Cognifide/knotx/wiki/Repository) feature.
```
Comment (tomaszmichalak) [inline README.md]: Knot.x can combine several template sources thanks to its Repository feature. It allows to have one entry point to different content platforms.

File: README.md [hunk 4]
```diff
@@ -3,118 +3,67 @@
...
+- **Knot.x** can render dynamic page that requires data from multiple external sources (e.g. microservices) thanks to [View Knot](https://github.com/Cognifide/knotx/wiki/ViewKnot) feature.
```
Comment (tomaszmichalak) [inline README.md]: **Knot.x** can also produce templates with dynamic data which comes from multiple external sources (e.g. microservices) thanks to [View Knot](https://github.com/Cognifide/knotx/wiki/ViewKnot) feature.

Knot.x uses performant and scalable [Vert.x](http://vertx.io/) engine as a heart of an architecture. Learn more about [Knot.x Architecture]

File: README.md (hunk 1 above)
Comment (Skejven) [inline README.md]: Right, moved.

File: README.md (hunk 2 above)
Comment (Skejven) [inline README.md]: Changed to:
`Knot.x assembles static and dynamic content from multiple sources to produce page with dynamic data in very performant manner.`

File: README.md (hunk 3 above)
Comment (Skejven) [inline README.md]: Chagned.

File: README.md (hunk 4 above)
Comment (Skejven) [inline README.md]: Updated

**Comment by tomaszmichalak** [comment]
approved

**Pull request merged**